Packs/AnsibleTower/Integrations/AnsibleTower/AnsibleTower.py

```diff
@@ -0,0 +1,522 @@
+    inventory_id = args.get('inventory_id')
+    credential_id = args.get('credential_id')
+    module_name = args.get('module_name')
+    module_args = args.get('module_args')
+    body = {"inventory": inventory_id,
+            "credential": credential_id,
+            "module_name": module_name,
+            "module_args": module_args}
+    url_suffix = 'ad_hoc_commands/'
+    response = client.api_request(method='POST', url_suffix=url_suffix, json_data=body, ok_codes=[201])
+    context_data = output_data(response)
+    command_id = context_data.get('id', '')
+    command_status = context_data.get('status', '')
+    headers = get_headers(context_data)
+    return CommandResults(
+        outputs_prefix='AnsibleAWX.AdhocCommand',
+        outputs_key_field='id',
+        outputs=response,
+        readable_output=tableToMarkdown(name=f'Ad hoc command - {command_id} status - {command_status}', t=context_data,
+                                        removeNull=True, headers=headers),
+        raw_response=response
+    )
+
+
+def relaunch_ad_hoc_command(client: Client, args: dict) -> CommandResults:
+    command_id = args.get("command_id")
+    url_suffix = f'ad_hoc_commands/{command_id}/relaunch/'
+    response = client.api_request(method='POST', url_suffix=url_suffix, ok_codes=[201])
+    context_data = output_data(response)
+    command_id = context_data.get('id', '')
+    command_status = context_data.get('status', '')
+    headers = get_headers(context_data)
+    return CommandResults(
+        outputs_prefix='AnsibleAWX.AdhocCommand',
+        outputs_key_field='id',
+        outputs=response,
+        readable_output=tableToMarkdown(name=f'Ad hoc command - {command_id} status - {command_status}', t=context_data,
+                                        removeNull=True, headers=headers),
+        raw_response=response
+    )
+
+
+def cancel_ad_hoc_command(client: Client, args: dict) -> CommandResults:
+    command_id = args.get("command_id")
+    url_suffix = f'ad_hoc_commands/{command_id}/cancel/'
+    try:
+        client.api_request(method='POST', url_suffix=url_suffix, empty_valid_codes=[202], return_empty_response=True)
+        return ad_hoc_command_status(client, args)
+    except DemistoException as e:
+        if e.res.status_code == 405:
+            response = client.api_request(method='GET', url_suffix=url_suffix)
+            error_msg = f"{response}. Error: {e.message}"
+            raise DemistoException(error_msg, e)
+        raise DemistoException(e)
+
+
+def ad_hoc_command_stdout(client: Client, args: dict) -> CommandResults:
+    print_output = True if args.get('print_output', 'True') == 'True' else False
+    text_filter = args.get('text_filter', '')
+    command_id = args.get('command_id', None)
+    url_suffix = f'ad_hoc_commands/{command_id}/stdout/'
+    params = {"format": "json"}
+    response = client.api_request(method='GET', url_suffix=url_suffix, params=params)
```

Comment:
how about `request_params`? - just to avoid confusion with the integration params.